modules/users/default.nix

```diff
@@ -94,6 +94,12 @@ in
   };
 
   config = {
+    assertions = [
+      {
+        assertion = cfg.users ? root -> (cfg.users.root.home == null || cfg.users.root.home == "/var/root");
+        message = "`users.users.root.home` must be set to either `null` or `/var/root`.";
+      }
+    ];
```

Comment:
No objection to this, but any reason for this particular safety check? I guess it probably breaks things a bunch to change this?

Reply:
The code that ensures `home` is the same between the config and the system is set to not run on `root`, so this is the only place that checks `root`'s `home` is correct

Reply:
I've added a comment to make the intention clearer and it should be more clear now that it's in a more relevant commit